src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AuditMessage.java

```diff
@@ -307,15 +307,11 @@ public void addRestParams(Map<String,String> params) {
 
     public void addRestHeaders(Map<String,List<String>> headers, boolean excludeSensitiveHeaders) {
         if(headers != null && !headers.isEmpty()) {
-            if(excludeSensitiveHeaders) {
-                final Map<String, List<String>> headersClone = new HashMap<String, List<String>>(headers)
-                        .entrySet().stream()
-                        .filter(map -> !map.getKey().equalsIgnoreCase(AUTHORIZATION_HEADER))
-                        .collect(Collectors.toMap(p -> p.getKey(), p -> p.getValue()));
-                auditInfo.put(REST_REQUEST_HEADERS, headersClone);
-            } else {
-                auditInfo.put(REST_REQUEST_HEADERS, new HashMap<String, List<String>>(headers));
+            final Map<String, List<String>> headersClone = new HashMap<>(headers);
+            if (excludeSensitiveHeaders) {
+                headersClone.keySet().removeIf(AUTHORIZATION_HEADER);
```

Comment:
```
headers = Maps.filterEntries(headers, header -> !AUTHORIZATION_HEADER.test(header.getKey()));
```

Reply:
What is the advantage of doing this ?

Reply:
There is no copy/clone, it is a read only filtered view on the original map.